src/runtime-rs/crates/runtimes/virt_container/src/lib.rs

```diff
@@ -104,7 +101,23 @@ impl RuntimeHandler for VirtContainer {
     }
 }
 
-async fn new_hypervisor(_toml_config: &TomlConfig) -> Result<Arc<dyn Hypervisor>> {
-    // TODO: implement ready hypervisor
-    todo!()
+async fn new_hypervisor(toml_config: &TomlConfig) -> Result<Arc<dyn Hypervisor>> {
+    let hypervisor_name = &toml_config.runtime.hypervisor_name;
+    let hypervisor_config = toml_config
+        .hypervisor
+        .get(hypervisor_name)
+        .ok_or_else(|| anyhow!("failed to get hypervisor for {}", &hypervisor_name))
+        .context("get hypervisor")?;
+
+    // TODO: support other hypervisor
+    match hypervisor_name.as_str() {
+        HYPERVISOR_DRAGONBALL => {
+            let mut hypervisor = Dragonball::new();
+            hypervisor
+                .set_hypervisor_config(hypervisor_config.clone())
+                .await;
+            Ok(Arc::new(hypervisor))
+        }
+        _ => Err(anyhow!("Unsupported hypervisor {}", &hypervisor_name)),
```

Comment:
It would be good to add a GitHub issue URL here for adding support for the existing hypervisors.

Reply:
Thanks for your review, rise an issue #4634.